docs/docs/how-tos/configure-keycloak-howto.md

```diff
@@ -0,0 +1,167 @@
+
+2. Click the **Add user** button and you will see the new user form:
+
+   ![Keycloak add user tab screenshot - new user form](/img/how-tos/keycloak_add_users.png)
+
+3. Fill out the three fields outlined above. These are **Username**, **Email**, and **Groups**. (See explanation below). Then click **save**.
+
+   - Username: Depending on the authentication provider selected ('password', 'GitHub' or 'Auth0'), the values entered into the **Username** field will differ slightly. The following table outlines
+     those differences:
+
+     |          | Password          | GitHub            | Auth0                 |
+     | -------- | ----------------- | ----------------- | --------------------- |
+     | Username | _unique username_ | _GitHub username_ | _Email to login with_ |
+
+- Once the **Username** field is updated, add a valid email address in the **Email** field.
+
+  :::note
+  Although not required, users may not be able to log in to Grafana if this field isn't properly set.
+  :::
+
+- Associate the user with one or more of the **Groups**. Out of the box, Nebari is deployed with the following groups: `admin`, `developer` and `analyst` (see the [Roles x Groups](#roles-x-groups) section below for more details).
+
+1. Click **Save**.
+
+If you are using the password authentication provider, you will also need to define a password for the user. It's best to put the **Temporary** toggle in the **OFF** position. Otherwise the user will be forced to change the password on first login.
+
+:::note
+If using Auth0, Github or any other identity provider, the password field is not required.
+:::
+
+![Keycloak add user > credentials tab screenshot - set password](/img/how-tos/keycloak_user_password.png)
+
+### Add user from the command line
+
+:::warning
+If you [changed the initial root password for Keycloak](#change-keycloak-root-password) this method will not work.
+:::
+
+To make adding users easier for new Nebari deployments, we've created a CLI command to help you.
+
+```shell
+nebari keycloak -c nebari-config.yaml adduser <username> <password>
+```
+
+This will create a new user `<username>` under the `analyst` group, with the initial password provided. Omit the password completely if you are using GitHub or Auth0. It will also add a dummy email (i.e. `username@your-domain`) to the **Email** field.
+
+## Login to Nebari
+
+Your new user can now log in to Nebari, visit your provided Nebari domain URI which will take you to the login form page and follow the [How to login in to Nebari and start a server](/how-tos/login-keycloak).
+
+![Nebari - Log in to Keycloak page](/img/how-tos/nebari_login_screen.png)
+
+## In-depth look at Roles and Groups
+
+Groups represent a collection of users that perform similar actions and therefore require similar permissions. By default, Nebari is deployed with the following groups: `admin`, `developer`, and `analyst` (in roughly descending order of permissions and scope).
```

Comment:
This "in-depth look" is a great section to have, thank you for writing it!

Can we also include a description of "Roles" here similar to what we have for "Groups"? -- I think I don't fully understand how the Realm role and client roles are connected, so maybe a sentence about that will also help. 

(This can be a part of future work as a part of our Glossary of terms as well)

Reply:
Definitely! I added a small paragraph trying to outline the differences but let me know if this is unclear 👍

Reply:
Thank you!